app/src/marketplace/utils/web3.js

```diff
@@ -2,24 +2,128 @@
 
 import BN from 'bignumber.js'
 import Web3 from 'web3'
+import { I18n } from 'react-redux-i18n'
 
-import { StreamrWeb3 } from '$shared/web3/web3Provider'
+import NoBalanceError from '$mp/errors/NoBalanceError'
+import { StreamrWeb3, getWeb3 } from '$shared/web3/web3Provider'
 import getConfig from '$shared/web3/config'
 import type { SmartContractCall } from '$shared/flowtype/web3-types'
 import { getContract, call } from '../utils/smartContract'
-
+import { gasLimits, paymentCurrencies } from '$shared/utils/constants'
 import { fromAtto } from './math'
+import type { PaymentCurrency } from '$shared/flowtype/common-types'
 
 declare var ethereum: Web3
 
-const tokenContractMethods = () => getContract(getConfig().token).methods
+const dataTokenContractMethods = () => getContract(getConfig().dataToken).methods
+const daiTokenContractMethods = () => getContract(getConfig().daiToken).methods
+const uniswapAdaptorMethods = () => getContract(getConfig().uniswapAdaptor).methods
 
 export const getEthBalance = (web3Instance: StreamrWeb3): Promise<number> => (web3Instance.getDefaultAccount()
     .then((myAccount) => web3Instance.eth.getBalance(myAccount).then((balance) => BN(balance)))
     .then(fromAtto).then((result) => result.toString())
 )
 
 export const getDataTokenBalance = (web3Instance: StreamrWeb3): SmartContractCall<number> => (web3Instance.getDefaultAccount()
-    .then((myAddress) => call(tokenContractMethods().balanceOf(myAddress)))
+    .then((myAddress) => call(dataTokenContractMethods().balanceOf(myAddress)))
+    .then(fromAtto).then((result) => result.toString())
+)
+
+export const getDaiTokenBalance = (web3Instance: StreamrWeb3): SmartContractCall<number> => (web3Instance.getDefaultAccount()
+    .then((myAddress) => call(daiTokenContractMethods().balanceOf(myAddress)))
     .then(fromAtto).then((result) => result.toString())
 )
+
+export const getBalances = (): Promise<[BN, BN, BN]> => {
+    const web3 = getWeb3()
+    const ethPromise = getEthBalance(web3)
+    const dataPromise = getDataTokenBalance(web3)
+    const daiPromise = getDaiTokenBalance(web3)
+
+    return Promise.all([ethPromise, dataPromise, daiPromise])
```

Comment:
`return Promise.all([ethPromise, dataPromise, daiPromise])` should be enough, they are already BigNumbers